packages/iov-bns-governance/src/index.ts

```diff
@@ -0,0 +1,137 @@
+import { As } from "type-tagger";
+
+import { Address, Amount, ChainId, PostTxResponse } from "@iov/bcp";
+import { BnsConnection, Electorate, Fraction, Proposal, VoteOption } from "@iov/bns";
+import { UserProfile } from "@iov/keycontrol";
+
+type CommitteeId = number & As<"committee">;
+
+/* Constants to be finalised once the committees are actually on-chain */
+export const COMMITTEE_IDS = {
+  ["testnet" as ChainId]: {
+    GOVERNING_BOARD: 1 as CommitteeId,
+    ECONOMIC_COMMITTEE: 2 as CommitteeId,
+    VALIDATOR_COMMITTEE: 3 as CommitteeId,
+    TECH_COMMITTEE: 4 as CommitteeId,
+  },
+};
+
+export enum ProposalType {
+  ADD_COMMITTEE_MEMBER,
+  REMOVE_COMMITTEE_MEMBER,
+  AMEND_COMMITTEE_THRESHOLD,
+  AMEND_COMMITTEE_QUORUM,
+  ADD_VALIDATOR,
+  REMOVE_VALIDATOR,
+  ACCESS_GUARANTEE_FUND,
+  DISTRIBUTE_FUNDS,
+  AMEND_PROTOCOL,
+}
+
+export interface AddCommitteeMemberOptions {
+  readonly type: ProposalType.ADD_COMMITTEE_MEMBER;
+  readonly committee: CommitteeId;
+  readonly address: Address;
+}
+
+export interface RemoveCommitteeMemberOptions {
+  readonly type: ProposalType.REMOVE_COMMITTEE_MEMBER;
+  readonly committee: CommitteeId;
+  readonly address: Address;
+}
+
+export interface AmendCommitteeThresholdOptions {
+  readonly type: ProposalType.AMEND_COMMITTEE_THRESHOLD;
+  readonly committee: CommitteeId;
+  readonly threshold: Fraction;
+}
+
+export interface AmendCommitteeQuorumOptions {
+  readonly type: ProposalType.AMEND_COMMITTEE_QUORUM;
+  readonly committee: CommitteeId;
+  readonly quorum: Fraction | null;
+}
+
+export interface AddValidatorOptions {
+  readonly type: ProposalType.ADD_VALIDATOR;
+  readonly address: Address;
+}
+
+export interface RemoveValidatorOptions {
+  readonly type: ProposalType.REMOVE_VALIDATOR;
+  readonly address: Address;
+}
+
+export interface AccessGuaranteeFundOptions {
+  readonly type: ProposalType.ACCESS_GUARANTEE_FUND;
+  readonly amount: Amount;
+}
+
+export interface DistributeFundsOptions {
+  readonly type: ProposalType.DISTRIBUTE_FUNDS;
+  readonly recipients: readonly {
+    readonly address: Address;
+    readonly weight: number;
+  }[];
+}
+
+export interface AmendProtocolOptions {
+  readonly type: ProposalType.AMEND_PROTOCOL;
+  readonly text: string;
+}
+
+export type ProposalOptions =
+  | AddCommitteeMemberOptions
+  | RemoveCommitteeMemberOptions
+  | AmendCommitteeThresholdOptions
+  | AmendCommitteeQuorumOptions
+  | AddValidatorOptions
+  | RemoveValidatorOptions
+  | AccessGuaranteeFundOptions
+  | DistributeFundsOptions
+  | AmendProtocolOptions;
+
+export interface GovernorOptions {
+  readonly connection: BnsConnection;
+  readonly profile: UserProfile;
+}
+
+export class Governor {
+  private readonly connection: BnsConnection;
+  private readonly profile: UserProfile;
+
+  public constructor({ connection, profile }: GovernorOptions) {
```

Comment:
The voting app will not have the full profile available, because all the keys are in the browser extension. I think we should pass the governor `Identity` instead